src/main/com/aca/homework/week7/clickable/AbstractClickable.java

```diff
@@ -0,0 +1,21 @@
+package com.aca.homework.week7.clickable;
+
+public abstract class AbstractClickable implements Clickable {
+
+    private Action action;
+
+    public AbstractClickable() {
+        this.action = new Action() {
```

Comment:
please do not hardwire the Action object here. instead, require it by constructor